demofile-parsing/Round.js

```diff
@@ -0,0 +1,163 @@
+const demofile = require("demofile");
+const asciichart = require('asciichart');
+
+class Round {
+
+  constructor (startTime, endTime, keyEvents) {
+    this.startTime = startTime || null;
+    this.endTime = endTime || null;
+    this.keyEvents = keyEvents || [];
+    this.eventRates = [];
+    this.highRateTimes = [];
+    this.highlights = [];
+  }
+
+  // check if a time is within the round
+  inRound(time) {
+    return this.startTime <= time && this.endTime >= time;
+  }
+
+  pushKeyEvent(keyEvent) {
+    this.keyEvents.push(keyEvent);
+  }
+
+  sortKeyEvents() {
+    this.keyEvents.sort(this.compare);
+  }
+
+  compare( a, b ) {
+    if ( a.time < b.time ){
+      return -1;
+    }
+    if ( a.time > b.time ){
+      return 1;
+    }
+    return 0;
+  }
+
+  calculateEventRates() {
+    var i;
```

Comment:
Pedantic - but this lonesome i outside the for loop is stressing me.

Reply:
the boyboy var i